lib/extract/netcdf/lfric/kernel_data_netcdf.jinja

```diff
@@ -0,0 +1,249 @@
+    !! @param[inout] this The instance of the extract_PsyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine DeclareFieldDouble(this, name, value)
+        use field_mod, only : field_type, field_proxy_type
+        implicit none
+
+        class(extract_PsyDataType), intent(inout), target :: this
+        character(*), intent(in)                          :: name
+        type(field_type), intent(in)                      :: value
+
+        type(field_proxy_type) :: value_proxy
+
+        value_proxy = value%get_proxy()
+        call this%ExtractNetcdfBaseType%DeclareArray1dDouble(name, value_proxy%data)
+    end subroutine DeclareFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine writes the value of a double precision LFRic field
+    !! to the NetCDF file. It uses the 1d-double precision array
+    !! writing function from the base class.
+    !! @param[inout] this The instance of the extract_PsyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine WriteFieldDouble(this, name, value)
+        use field_mod, only: field_type, field_proxy_type
+        implicit none
+
+        class(extract_PsyDataType), intent(inout), target :: this
+        character(*), intent(in)                          :: name
+        type(field_type), intent(in)                      :: value
+
+        type(field_proxy_type) :: value_proxy
+
+        value_proxy = value%get_proxy()
+        call this%ExtractNetcdfBaseType%WriteArray1dDouble(name, value_proxy%data)
+    end subroutine WriteFieldDouble
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine declares an LFric field vector. Each component of the
+    !! vector is stored as a separate variable, using the 1d-double precision
+    !! array function of the base class.
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] name The name of the variable (string).
+    !! @param[in] value The value of the variable.
+    subroutine DeclareFieldVectorDouble(this, name, value)
+        use netcdf, only: nf90_def_dim, nf90_def_var, NF90_REAL8
+        use field_mod, only : field_type, field_proxy_type
+        implicit none
+
+        class(extract_PSyDataType), intent(inout), target :: this
+        character(*), intent(in)                          :: name
+        type(field_type), dimension(:), intent(in)        :: value
+
+        integer                :: i
+        type(field_proxy_type) :: value_proxy
+        character(9)           :: number
+
+        ! Provide each component of the vector as an individual 1d array.
+        ! The base class will re-allocate internal array sizes if required
+        do i = 1, size(value)
+            value_proxy = value(1)%get_proxy()
```

Comment:
Shouldn't that be `value(i)`?

Reply:
Indeed. I've opened #991 to add testing for the PSyData libraries.